enhancements/sig-policy/98-long-term-compliance-history/README.md

```diff
@@ -0,0 +1,506 @@
+endpoint would be protected with authentication and the credentials would be stored in a secret called
+`governance-policy-framework-cache` in the managed cluster namespace. Each policy controller would read this secret on
+start up. Recording the compliance event directly from the controller would allow recording rich metadata about the
+compliance event. One example is the diff between what is in the policy and what is on the cluster, similar to what
+ArgoCD does.
+
+The Governance Policy Framework would still have a goroutine that periodically polls etcd for compliance events while
+disconnected from the hub, however, when there is a connection to the hub, it can transition to watching for new keys
+with the prefix `compliance-events/`.
+
+### User Stories
+
+#### Story 1 - Phase 1
+
+As a policy user, I require the compliance history of my fleet of clusters so that I can perform an audit over the last
+two years.
+
+#### Story 2 - Phase 2
+
+As a policy user, I require that compliance history events still be recorded when the cluster is disconnected from the
+hub.
+
+#### Story 3
+
+As a policy user, I'd like an easy way to generate a spreadsheet report of policy compliance events within a date range
+to make auditing easier.
+
+#### Story 4 - Phase 3 (Optional)
+
+As a policy user, I require that compliance history events contain additional metadata about the violation such as the
+diff between what is declared in the policy and what is on the cluster to know what the exact violation was. This will
+help in forensic investigations after a breach has occurred of a managed cluster.
+
+#### Story 5
+
+As a policy user, I'd like to import my existing compliance history to the new compliance events database.
+
+### Implementation Details/Notes/Constraints [optional]
+
+#### Embedded etcd Resource Utilization
+
+The following minimal embedded etcd Go program uses around 3MB of memory for the etcd portion under light load:
+
+```go
+package main
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"time"
+
+	"go.etcd.io/etcd/server/v3/embed"
+)
+
+func main() {
+	cfg := embed.NewConfig()
+	cfg.AdvertiseClientUrls = []url.URL{}
+	cfg.AdvertisePeerUrls = []url.URL{}
+	cfg.ListenPeerUrls = []url.URL{}
+	cfg.ListenMetricsUrls = []url.URL{}
+	cfg.EnableV2 = false
+	cfg.Dir = "default.etcd"
+	e, err := embed.StartEtcd(cfg)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer e.Close()
+	select {
+	case <-e.Server.ReadyNotify():
+		log.Printf("Server is ready!")
+	case <-time.After(60 * time.Second):
+		e.Server.Stop() // trigger a shutdown
+		log.Printf("Server took too long to start!")
+	}
+	log.Fatal(<-e.Err())
+}
+```
+
+### Risks and Mitigation
+
+### Open Questions [optional]
+
+1. How to enable it on the managed cluster? Should it be per managed cluster?
+1. Is the RBAC sufficient? Instead of "get" access on a managed cluster, should we have a custom verb of
```

Comment:
That's a good point on deleted clusters. I think you could still have a custom verb since `resourceNames` would either be not set or they would include the deleted clusters.

Reply:
This can be further clarified in the threat model